onnxscript/ir/_core.py

```diff
@@ -2605,15 +2603,12 @@ def __init__(
         # and not from an outer scope
         graph: Graph,
         attributes: Sequence[Attr],
-        metadata_props: dict[str, str] | None = None,
```

Comment:
I would preserve this line. In the ref commit there is a line of logic that calls dict.update